### ISSUE TITLE
Linting disappeares when using multiple folders in workspace

### ISSUE THREAD

**Issue by Bendi06** [microsoft/pylance-release#3652]
## Environment data

-   Language Server version: 2022.11.30 (pyright 04a10583)
-   OS and version: Windows 10 Pro 21H2 19044.2251
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.8 64-bit, not python version dependent bug

## Code Snippet

[VS Code Workspace Python bug.zip](https://github.com/microsoft/pylance-release/files/10044510/VS.Code.Workspace.Python.bug.zip)

## Repro Steps

1. Create two folders, add a python file in each of them.
2. Open one folder with VS Code and add the other to the workspace.
3. (Sometimes not needed) Restart VS Code, reopen the same folder.
4. Type anything into the file in the main folder. Linting will be gone.

## Expected behavior

Linting does not disappear after opening other folders in workspace.

## Actual behavior

After adding a folder to workspace, linting disappears (maybe language server crashes).

## Logs
[Python Language Server Output.txt](https://github.com/microsoft/pylance-release/files/10044534/Python.Language.Server.Output.txt)

```
Video of the problem:
https://user-images.githubusercontent.com/68500992/202795016-38dd2e4d-4832-4963-9719-a9296cecbc8d.mp4


```

**Comment by Torom** [comment]
+1 Exactly the same behavior occurs for me.

-   Language Server version: 2022.11.30 (pyright 04a10583)
-   OS and version: Arch Linux 6.0.9-arch1-1
-   Python version: 3.11.0

**Comment by rchiodo** [comment]
This is strange:

```
[Info  - 21:03:27] (13772) Searching for source files
[Info  - 21:03:27] (13772) No source files found.
```

Can you include the contents of your user settings.json? For some reason we don't think there are any workspaces open.

**Comment by erictraut** [comment]
@rchiodo, the "No source files found" is only for the "<default>" service instance. That makes sense because we use the "<default>" service instance for files that don't appear within any of the workspace roots — some other file that you happen to open in your editor. The logs show that there were are two workspaces called "Main" and "Other Project". We created one service instance for each, and one source file was found under each workspace.

**Comment by Bendi06** [comment]
> This is strange:
> 
> ```
> [Info  - 21:03:27] (13772) Searching for source files
> [Info  - 21:03:27] (13772) No source files found.
> ```
> 
> Can you include the contents of your user settings.json? For some reason we don't think there are any workspaces open.

Here is the settings.json (.txt because github does not support json for some reason):
[settings.json.txt](https://github.com/microsoft/pylance-release/files/10059304/settings.json.txt)

**Comment by rchiodo** [comment]
Thanks. Can you turn on trace logging:

![image](https://user-images.githubusercontent.com/19672699/203129567-6bfa6629-fd94-4f43-9171-5ba1e80413fb.png)

And try again? 

I don't repro the problem. Semantic tokens are working for me on reload. 

This should appear in the output log:

```
(23176) [BG(1)] getSemanticTokens range 0:0 - 7:1 at d:\Source\Testing\VS Code Workspace Python bug\Main\main.py (169ms)
(23176) [BG(3)]     parsing: c:\Users\aku91\.vscode\extensions\ms-python.vscode-pylance-2022.11.32\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 1ms] (23ms)
(23176) [BG(3)]     binding: c:\Users\aku91\.vscode\extensions\ms-python.vscode-pylance-2022.11.32\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (1ms)
(23176) [BG(3)]     parsing: c:\Users\aku91\.vscode\extensions\ms-python.vscode-pylance-2022.11.32\dist\typeshed-fallback\stdlib\types.pyi [fs read 0ms] (6ms)
(23176) [BG(3)]     binding: c:\Users\aku91\.vscode\extensions\ms-python.vscode-pylance-2022.11.32\dist\typeshed-fallback\stdlib\types.pyi (2ms)
(23176) [BG(3)]     parsing: c:\Users\aku91\.vscode\extensions\ms-python.vscode-pylance-2022.11.32\dist\typeshed-fallback\stdlib\abc.pyi [fs read 1ms] (1ms)
(23176) [BG(3)]     binding: c:\Users\aku91\.vscode\extensions\ms-python.vscode-pylance-2022.11.32\dist\typeshed-fallback\stdlib\abc.pyi (1ms)
(23176) [BG(3)]   checking: d:\Source\Testing\VS Code Workspace Python bug\Main\main.py (100ms)
(23176) [BG(3)] analyzing: d:\Source\Testing\VS Code Workspace Python bug\Main\main.py (210ms)
```

Hopefully if that's failing, we'll have some output about the failure.

**Comment by rchiodo** [comment]
Sorry my bad, I skipped a step.

`Type anything into main.py and the colors will go away`.

That caused the repro. And `getSemanticTokens` is failing.

**Comment by Bendi06** [comment]
> Sorry my bad, I skipped a step.
> 
> `Type anything into main.py and the colors will go away`.
> 
> That caused the repro. And `getSemanticTokens` is failing. So we should be able to debug it here. Thanks.

No problem, if you need anything I'm happy to help.

**Comment by rchiodo** [comment]
Debugging it, it looks like the sourceFile has empty contents during the textDocument/didChange event. Not sure why yet.

**Comment by rchiodo** [comment]
Initial guess is that the adding of the new folder causes a `textDocument/didOpen` to be handled by the 'default' workspace, and not the workspace that will eventually handle the `textDocument/didChange` event.

**Comment by OseiasALira** [comment]
- Language Server version: 2022.11.30 (pyright 04a10583)
- OS and version: Windows 11 Pro 22H2 22621.819
- Python version: 3.11.0

Same issues with me.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221141-30-November-2022-PreRelease)
